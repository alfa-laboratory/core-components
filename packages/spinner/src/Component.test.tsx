import React from 'react';
import { render } from '@testing-library/react';

import { Spinner } from './index';

const testId = 'spinner';

describe('Snapshots tests', () => {
    it('should display correctly', () => {
        const { container } = render(<Spinner />);

        expect(container).toMatchSnapshot();
    });
});

describe('Attributes tests', () => {
    it('should set data-test-id attribute', async () => {
        const className = 'custom';

        render(<Spinner className={className} dataTestId={testId} />);

        const spinnerContentWrap = document.querySelector(`.${className}`);

        const testIdAttr = spinnerContentWrap?.getAttribute('data-test-id');

        expect(testIdAttr).toBe(testId);
    });
});

describe('Render tests', () => {
    it('should unmount without errors', async () => {
        const { unmount } = render(<Spinner />);

        expect(unmount).not.toThrowError();
    });

    it('should have visible class if prop visible is true', async () => {
        const { getByTestId } = render(<Spinner dataTestId={testId} visible={true} />);

        expect(getByTestId(testId)).toHaveClass('visible');
    });

    it('should set `size` class', () => {
        const size = 's';
        const { container } = render(<Spinner visible={true} size={size} />);

        expect(container.firstElementChild).toHaveClass(size);
    });

    it('should use default `size`', () => {
        const { container } = render(<Spinner visible={true} />);

        expect(container.firstElementChild).toHaveClass('s');
    });
});
