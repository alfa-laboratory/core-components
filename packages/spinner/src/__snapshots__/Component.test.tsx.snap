// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots tests should display correctly 1`] = `
<div>
  <span
    class="spinner s"
  />
</div>
`;
